### PR TITLE
Sonic Robo Blast 2: Add recipes

### PR DESCRIPTION
**Pull request by win8linux**
[*Sonic Robo Blast 2* (SRB2)](https://www.srb2.org/) is an open-source *Sonic the Hedgehog* fangame that attempts to translate the classic Sonic experience from the Sega Genesis games to 3D using a modified *Doom Legacy* port of *Doom*.

The recipe currently uses a fork based on the 2.2.9 release with some Haiku fixes (special thanks to @mmuman):
<https://git.do.srb2.org/win8linux/SRB2/-/tree/haiku_release>

There is a branch in that fork which is based on `next`, however a platform detection issue prevents it from compiling:
<https://git.do.srb2.org/win8linux/SRB2/-/tree/haiku_release>
If that can be resolved through a fix or compile flag, the process of upstreaming better Haiku support can begin.

In any case, the game does work with the Haiku fixes:
srb2_screenshot1
srb2_screenshot5
srb2_screenshot6

[More screenshots here](https://discuss.haiku-os.org/t/openmw-morowind-and-other-3d-games/7363/93)

**NOTE:** The game has a network check during startup that may sometimes take a long time to finish, depending on network conditions on the user's side and that of the master server (for multiplayer). Compiling with either `NONET=1` or `NOCURL=1` results in building and/or linking errors, unfortunately. If the game gets stuck at the network check, then try another time or leave it running until it finishes (did the latter eventually and it worked).

### CODE REVIEW

**Comment by win8linux** [comment]
Went into the SRB2 Discord to ask about redistributing game assets and one of the developers said:
> I don't think the assets should be packed, the source makes more sense since gpl (unsure)

They're only going off based on what they remember, though. Regardless, the ambiguity is not ideal and the data package will be removed for the meantime. It can be added back another time, should it later turn out that it can be redistributed in a package.

**Comment by win8linux** [comment]
Should the game package description mention that data files can be put anywhere in the home folder?

**Review by Begasus — APPROVED** [review]

File: games-arcade/srb2/srb2-2.2.9.recipe [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	devel:libexecinfo$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libopenmpt$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+	devel:libSDL2_mixer_2.0$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
```
[the remark this answers is not in the document]
Comment (pulkomandy) [reply games-arcade/srb2/srb2-2.2.9.recipe:46]: So please remove the useless commands from the BUILD_PREREQUIRES: cmake is useless and I think git is probably also useless.

[the remark this answers is not in the document]
Comment (win8linux) [reply games-arcade/srb2/srb2-2.2.9.recipe:46]: I could remove `cmake`, but not `git`. It is used by the Makefile during the build process.

**Pull request merged**